pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -161,3 +164,179 @@ func (s *Service) newQueryMux() *datasource.QueryTypeMux {
 	}
 	return mux
 }
+
+func (s *Service) getDSInfo(pluginCtx backend.PluginContext) (types.DatasourceInfo, error) {
+	i, err := s.im.Get(pluginCtx)
+	if err != nil {
+		return types.DatasourceInfo{}, err
+	}
+
+	instance, ok := i.(types.DatasourceInfo)
+	if !ok {
+		return types.DatasourceInfo{}, fmt.Errorf("failed to cast datsource info")
+	}
+
+	return instance, nil
+}
+
+func checkAzureMonitorMetricsHealth(dsInfo types.DatasourceInfo) (*http.Response, error) {
+	url := fmt.Sprintf("%v/subscriptions?api-version=2019-03-01", dsInfo.Routes["Azure Monitor"].URL)
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := dsInfo.Services["Azure Monitor"].HTTPClient.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func checkAzureLogAnalyticsHealth(dsInfo types.DatasourceInfo) (*http.Response, error) {
+	defaultWorkspaceId := dsInfo.Settings.LogAnalyticsDefaultWorkspace
+
+	if defaultWorkspaceId == "" {
+		workspacesUrl := fmt.Sprintf("%v/subscriptions/%v/providers/Microsoft.OperationalInsights/workspaces?api-version=2017-04-26-preview", dsInfo.Routes["Azure Monitor"].URL, dsInfo.Settings.SubscriptionId)
+		workspacesReq, err := http.NewRequest(http.MethodGet, workspacesUrl, nil)
+		if err != nil {
+			return nil, err
+		}
+		res, err := dsInfo.Services["Azure Monitor"].HTTPClient.Do(workspacesReq)
+		if err != nil {
+			return nil, err
+		}
+		var target struct {
+			Value []types.LogAnalyticsWorkspaceResponse
+		}
+		err = json.NewDecoder(res.Body).Decode(&target)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(target.Value) == 0 {
+			return nil, errors.New("no default workspace found")
```

Comment:
I personally don't think this case should be an error but rather a warning to the user that there is no available log analytics workspace. I'd like thoughts on this as currently the `BackendCheckHealthResult` status only supports `UNKNOWN`, `ERROR`, `OK`. I think an additional warning status should be included allowing for a warning alert to be displayed in the frontend rather than an error.

Reply:
I will test this and get back to you 😊